test/functional/eval/timer_spec.lua

```diff
@@ -220,16 +220,20 @@ describe('timers', function()
       :good^                                   |
     ]])
 
+    -- Wait for timer to have finished / echoed.
+    retry(nil, nil, function()
+      eq(1, eval('g:val'))
+    end)
+
+    -- "evil" was not printed.
     screen:expect{grid=[[
                                               |
       {0:~                                       }|
       {0:~                                       }|
       {0:~                                       }|
       {0:~                                       }|
       :good^                                   |
-    ]], intermediate=true, timeout=1000}
-
-    eq(1, eval('g:val'))
+    ]]}
```

Comment:
Looks the same as the previous screen test. Will it warn about "unchanged" now?

Reply:
Oh, true:

> warning: Screen test succeeded immediately. …

Keeping `intermediate=true` fixes it.

We might keep the initial timeout then now also I assume (to ensure it is not changed for 200ms still), and the extra check added (waiting for the timer to have finished) should then fix the initial flakiness on osx.

Reply:
> Keeping `intermediate=true` fixes it.

I would think `unchanged=true` should be used. Perhaps `intermediate=true` works by accident...

Reply:
`unchanged=true` fails with intermediate states.

Using `screen:redraw_debug()` shows:
```
screen:expect{grid=[[
                                          |
  {0:~                                       }|
  {0:~                                       }|
  {0:~                                       }|
  {0:~                                       }|
  :good^                                   |
]]}

{ "grid_line", { 1, 5, 0, { { "e", 0 }, { "v" }, { "i" }, { "l" } } }, { 1, 5, 4, { { " ", 0, 1 } } } }
{ "grid_cursor_goto", { 1, 5, 4 } }
{ "flush", {} }

screen:expect{grid=[[
                                          |
  {0:~                                       }|
  {0:~                                       }|
  {0:~                                       }|
  {0:~                                       }|
  evil^                                    |
]]}

{ "grid_line", { 1, 5, 0, { { ":", 0 } } }, { 1, 5, 1, { { "g", 0 }, { "o", 0, 2 }, { "d" } } } }
{ "grid_cursor_goto", { 1, 5, 5 } }
{ "flush", {} }

screen:expect{grid=[[
                                          |
  {0:~                                       }|
  {0:~                                       }|
  {0:~                                       }|
  {0:~                                       }|
  :good^                                   |
]]}
```

i.e. "evil" shows up, but get overwritten then apparently.  Asserting this is flaky though (i.e. not seen all the time)

@bfredl 
Is that how it should work?  Or should "evil" never be echoed in the first place really?

Reply:
@blueyed If "evil" was not echoed then this test would have tested nothing at all. (though an optimization to suppress it wouldn't be wrong by itself, but then we would need to change the test to use something else that is not). The purpose is to check whether the command line is restored (by nvim itself) after being "messed up" by a timer.  Using `intermediate=true` is not somehow an "accident", it is used because it is exactly what the test is supposed to check.